docs/authoring.md

```diff
@@ -259,3 +224,118 @@ There are some template parameters that have a special meaning:
 |Name|Description|
 |---|---|
 |`server`| It is used to let the template know which server from the AsyncAPI specification file you want to use. In some cases, this may be required. For instance, when generating code that connects to a specific server. Use this parameter in case your template relies on users' information about what server from the specification file they want to use during generation. You also need this parameter if you want to use `server.protocol` notation within `conditionalFiles` configuration option. Once you decide to specify this parameter for your template, it is recommended you make it a mandatory parameter otherwise feature like `conditionalFiles` is not going to work if your users do not use this parameter obligatory.
+
+
+## React
+
+[React](https://reactjs.org) is the render engine that we strongly suggest you should use for any new templates. The only reason it is not the default render engine is to stay backward compatible.
+
```

Comment:
please add also an important point about components reusability that we could not achieve with nunjucks (like share template file across code templates for the same language)

Reply:
@derberg but thats also possible with nunjucks right? You just have to point to the direct source in node_modules when installed as a dependency 🤔 ?

Reply:
oh, you mean like add a template to dependency and then point to it directly, like when importing macro? Technically it works, but it is error-prone, you kinda hardcode yourself to a template file structure, we would never go this way

Reply:
also, components testability improves how we can test template's output

Reply:
Does not have to be a template dependency, could be a library containing nunjucks partials in some directory. Not quite sure how the generator behaves in regards to outside partials. however do you still think we should add the reason since it is already stated right below in the list 🤔 ?

Reply:
> also, components testability improves how we can test template's output

True, this is missing.

Reply:
I think so, because anything else with nunjucks would always be dependent on files structure. So if in a possible library of reusable macros, you would change folder structure, that would be a breaking change, something you do not want and it is easy to forget as it doesn't feel natural. This is in my opinion why we never even considered doing this ourselves and promoting such support.

Because we use React, we do not have such a problem.

Reply:
@derberg added the part about testability also the part about reusability is also stated here -> https://github.com/asyncapi/generator/blob/775d1ea6032fc9e403bb23afd100f317771af9f3/docs/authoring.md#react